### ISSUE TITLE
Image mean operation ?

### ISSUE THREAD

**Issue by china56321** [ultralytics/yolov5#833]
Did yolov5 do the operation of Mean image  subtraction  ?

**Comment by glenn-jocher** [comment]
@china56321 no.

**Comment by Aktcob** [comment]
@glenn-jocher Did u compare the map of Mean=0 with map of Mean=0.5(for example)?

**Comment by glenn-jocher** [comment]
@Aktcob a long time ago yes, I tried various input normalizations, including a full batchnorm layer on the inputs, but was surprised to see little to no effect on the results. I surmised that the existing batchnorm layers must already be doing this task effectively. In any case, you can try this yourself if you'd like.

You can insert a unity mean and variance normalization step using imagenet stats:
`mean=[0.485, 0.456, 0.406] and std=[0.229, 0.224, 0.225]`

https://forums.fast.ai/t/is-normalizing-the-input-image-by-imagenet-mean-and-std-really-necessary/51338

**Comment by glenn-jocher** [comment]
Also, FYI you can see your layer stats (mean and variance) by using `model.info(verbose=True)`. You want your var to remain steady throughout the model to avoid vanishing gradients.

<img width="1208" alt="Screen Shot 2020-08-25 at 12 37 01 AM" src="https://user-images.githubusercontent.com/26833433/91145976-1bc4e880-e66b-11ea-9f6a-60a9269f0aa6.png">

**Comment by Aktcob** [comment]
@glenn-jocher pretty work! It seemed that no effect on the result when I train my own datasets. And removing the normalizations layers get more FPS.

**Comment by glenn-jocher** [comment]
@Aktcob ah ok. BN layers are removed during model.fuse() typically before inference/testing/exporting.

**Comment by Aktcob** [comment]
yep, I know it. I used TensorRT to do confusion with (Conv, bn). I did not use ur repo directly. But I used many tricks from ur repo. Really good job!~~~~~
